modules/twinklespeedy.js

```diff
@@ -998,126 +1075,96 @@ Twinkle.speedy.normalizeHash = {
-	'nopermission': 'No evidence of permission',
-// Categories
-	'catempty': 'Empty category',
-// User pages
-	'userreq': 'User request to delete page in own userspace',
-	'nouser': 'Userpage or subpage of a nonexistent user',
-	'gallery': '[[WP:NFC|Non-free]] [[Help:Gallery|gallery]]',
-	'notwebhost': '[[WP:NOTWEBHOST|Misuse of Wikipedia as a web host]]',
-// Templates
-	'policy': 'Template that unambiguously misrepresents established policy',
-	'duplicatetemplate': 'Unused, redundant template',
-// Portals
-	'p1': '[[WP:P|Portal]] page that would be subject to speedy deletion as an article',
-	'emptyportal': '[[WP:P|Portal]] without a substantial topic base',
-// Redirects
-	'rediruser': 'Cross-[[WP:NS|namespace]] [[WP:R|redirect]] from mainspace',
-	'redirtypo': 'Recently created, implausible [[WP:R|redirect]]'
-};
-
 Twinkle.speedy.callbacks = {
-	sysop: {
-		main: function( params ) {
-			var thispage;
+	getTemplateCodeAndParams: function(params) {
+		var code, parameters, i;
+		if (params.normalizeds.length > 1) {
+			code = "{{db-multiple";
+			params.utparams = {};
+			$.each(params.normalizeds, function(index, norm) {
+				code += "|" + norm.toUpperCase();
+				parameters = params.templateParams[index] || [];
+				for (var i in parameters) {
+					if (typeof parameters[i] === 'string' && !parseInt(i, 10)) {  // skip numeric parameters - {{db-multiple}} doesn't understand them
+						code += "|" + i + "=" + parameters[i];
+					}
+				}
+				$.extend(params.utparams, Twinkle.speedy.getUserTalkParameters(norm, parameters));
+			});
+			code += "}}";
+		} else {
+			parameters = params.templateParams[0] || [];
+			code = "{{db-" + params.values[0];
+			for (i in parameters) {
+				if (typeof parameters[i] === 'string') {
+					code += "|" + i + "=" + parameters[i];
+				}
+			}
+			if (params.usertalk) {
+				code += "|help=off";
+			}
+			code += "}}";
+			params.utparams = Twinkle.speedy.getUserTalkParameters(params.normalizeds[0], parameters);
+		}
 
-			Morebits.wiki.addCheckpoint();  // prevent actionCompleted from kicking in until user interaction is done
+		return [code, params.utparams];
+	},
 
-			// look up initial contributor. If prompting user for deletion reason, just display a link.
-			// Otherwise open the talk page directly
-			if( params.openusertalk ) {
-				thispage = new Morebits.wiki.page( mw.config.get('wgPageName') );  // a necessary evil, in order to clear incorrect status text
-				thispage.setCallbackParameters( params );
-				thispage.lookupCreator( Twinkle.speedy.callbacks.sysop.openUserTalkPage );
-			}
+	parseWikitext: function(wikitext, callback) {
+		var query = {
+			action: "parse",
+			prop: "text",
+			pst: "true",
+			text: wikitext,
+			title: mw.config.get("wgPageName")
+		};
+
+		var statusIndicator = new Morebits.status( 'Building deletion summary' );
+		var api = new Morebits.wiki.api( 'Fetching parsing deletion template', query, function(apiObj) {
```

Comment:
Haha, this and the comment above about `is` are a result of working on this during wee hours of the night